composeApp/src/commonMain/kotlin/org/ooni/probe/ui/run/RunScreen.kt

```diff
@@ -0,0 +1,238 @@
+    ) {
+        TopAppBar(
+            title = { Text(stringResource(Res.string.Dashboard_RunTests_Title)) },
+            navigationIcon = {
+                IconButton(onClick = { onEvent(RunViewModel.Event.BackClicked) }) {
+                    Icon(
+                        Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = stringResource(Res.string.back),
+                    )
+                }
+            },
+        )
+
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp)
+                .padding(bottom = 16.dp),
+        ) {
+            Text(
+                stringResource(Res.string.Dashboard_RunTests_Description),
+                style = MaterialTheme.typography.bodyLarge,
+            )
+            Row {
+                OutlinedButton(onClick = { onEvent(RunViewModel.Event.SelectAllClicked) }) {
+                    Text(stringResource(Res.string.Dashboard_RunTests_SelectAll))
+                }
+                OutlinedButton(onClick = { onEvent(RunViewModel.Event.DeselectAllClicked) }) {
+                    Text(stringResource(Res.string.Dashboard_RunTests_SelectNone))
+                }
+            }
+        }
+
+        Box {
+            LazyColumn(
+                contentPadding = PaddingValues(
+                    start = 16.dp,
+                    end = 16.dp,
+                    // Insets + Run tests button
+                    bottom =
+                        WindowInsets.navigationBars.asPaddingValues().calculateBottomPadding() +
+                            64.dp,
+                ),
+            ) {
+                val allSectionsHaveValues = state.list.entries.all { it.value.any() }
+                state.list.forEach { (type, descriptorsMap) ->
+                    if (allSectionsHaveValues && descriptorsMap.isNotEmpty()) {
+                        item(type) {
+                            TestDescriptorSection(type)
+                        }
+                    }
+
+                    descriptorsMap.forEach { (descriptorItem, testItems) ->
+                        val descriptor = descriptorItem.item
+                        item(descriptor.key) {
+                            DescriptorItem(
+                                descriptorItem = descriptorItem,
+                                onDropdownToggled = {
+                                    onEvent(RunViewModel.Event.DescriptorDropdownToggled(descriptor))
+                                },
+                                onChecked = {
+                                    onEvent(RunViewModel.Event.DescriptorChecked(descriptor, it))
+                                },
+                            )
+                        }
+
+                        if (!descriptorItem.isExpanded) return@forEach
+
+                        items(testItems, key = { it.item.test.name }) { testItem ->
```

Comment:
Cannot use name as `key` here. Test names can be repeated, already the case with `NMS` failing with 

```log
java.lang.IllegalArgumentException: Key "web_connectivity" was already used. If you are using LazyColumn/Row please make sure you provide a unique key for each item.
```

My suggestion is to use. name for tests not part of a descriptor and `descriptor_id_${test_name}` when possible.

Reply:
I think we should make provisions for customizing the items based on the flavor, `NMS` displays the list of inputs instead of the test names here.

Also the items dont present with a checkbox

Reply:
Additionally, the checkboxes are now presented on the left. See https://m3.material.io/components/checkbox/guidelines